Add tracing logging

JulianGCalderon opened the pull request:
Adds tracing logging to the replay binary. The logs have the following structure:
- A block span for all transactions in a given block
- A transaction span for the execution of a single transaction
- An error event is emitted if the status of the RPC and the blockifier execution don't match, or if the transaction fails unexpectedly
- An info event is emitted if the status matches

<img width="1440" alt="image" src="https://github.com/lambdaclass/starknet-replay/assets/60768809/c3e3baa7-6b2c-4058-a0ef-9d007ec82b72">

edg-l commented on replay/src/main.rs:
```diff
@@ -280,47 +277,44 @@ fn show_execution_data(
-            actual_fee.amount, actual_fee.unit
-        );
-        println!("[Blockifier] Actual fee: {:?} wei", blockifier_actual_fee);
+    // let execution_gas = execution_info.actual_fee;
```
you have commented out code here

JulianGCalderon replied:
Thanks for reminding me, I originally left it commented out as it was unclear if it was usefull information, but I changed it to a debug event instead.

pefontana reviewed (changes requested):
Nice one @JulianGCalderon !
Just two things:
* Can you add some lines in the README.md explaining how to filter the log messages
* Can we also add a rpc_execution_status field to the Error message?

JulianGCalderon commented:
@pefontana I added to the README how to see only error messages, but I couldn't find how to filter only reverted transactions.

I also added `rpc_execution_status` to the execution result event
